### PR TITLE
(re) example of correction to apply in order to avoid conflict during MNXref conversion

### PR DESCRIPTION
**Pull request by ANiknejad**
(sorry, I closed the previous pull request, because I thought I can choose the commits I want to show you, but now I have understand how it works...(I hope so ;-) ; anyway sorry for lot of pointless commits)

Hello,

I show you here an example of the correction we can do on the 'metabolites.tsv' file,
thanks to the information retrieved using the status codes generated during the conversion of the 'metatlas_HumanGEM' model to MNXref.

See the public documentation here:
https://github.com/MetaNetX/MNXtools/blob/main/doc/convert_manual.md

Here the case where the source id has conflicting xrefs, one xref is for a precise structure, the other is for a generic structure: depending on the source id, one should be removed

MetaNetX report file, we got

CHEM_XREF_CONFLICT MAM00097 (5-L-glutamyl)-L-amino acid
keggC:C03740 MNXM730664 (generic structure)
biggM:gluala MNXM1105815 (precise structure)
UNK:MAM00097

MNXM1105815 = chebi:50619 L-gamma-glutamyl-L-alanine
MNXM730664 = chebi:71304 an alpha-(gamma-L-glutamyl)-L-amino acid

First to do: check the name and structure and associated reactions for the source compound, here MAM00097 (5-L-glutamyl)-L-amino acid
https://metabolicatlas.org/explore/Human-GEM/gem-browser/metabolite/MAM00097s

Here the name is for a generic structure (-L-amino acid) but the associated chebi and the associated reactions are those for the defined
L-gamma-glutamyl-L-alanine

in the HUMAN_GEM 'metabolites.tsv' file, MAM00097s [(5-L-glutamyl)-L-amino acid] has xref: "gluala" "C03740" "HMDB06248" "CHEBI:50619'

gluala (bigg) is patched to chebi:50619 (L-gamma-glutamyl-L-alanine) and the xref here is correct, to report the compound that does the described reactions
C03740 (kegg) is patched to chebi:71304 (an alpha-(gamma-L-glutamyl)-L-amino acid)) and this structure is the generic form: we have to remove the xref
hmdb:HMDB0006248 is patched to chebi:50619 (L-gamma-glutamyl-L-alanine) and the xref here is correct, ok
CHEBI:50619: the precise structure, ok

to resolve this conflict:

by removing xref: C03740 (done in this commit)

I hereby confirm that I have:

- [X] Selected develop as a target branch

### CODE REVIEW

**Comment by ANiknejad** [comment]
hello Team,
@haowang-bioinfo 

May I ask you to review this PR? I am sorry in advance, many commits for at the end few modifications into the metabolites.tsv file, but anyway these changes should improve the integration of HumanGEM into MNXref common namespace.

Best regards,

Anne

File: model/metabolites.tsv [hunk 1]
```diff
@@ -183,7 +183,7 @@ mets	metsNoComp	metBiGGID	metKEGGID	metHMDBID	metChEBIID	metPubChemID	metLipidMa
 "MAM00095p"	"MAM00095"	"c226coa"	"C16169"	""	"CHEBI:65132"	""	"LMFA07050050"	"CE4827"	""	"CE4827;c226coa"	"MNXM3234;MNXM91778"	"m00095p"	"m00095p"
 "MAM00095r"	"MAM00095"	"c226coa"	"C16169"	""	"CHEBI:65132"	""	"LMFA07050050"	"CE4827"	""	"CE4827;c226coa"	"MNXM3234;MNXM91778"	"m00095r"	"m00095r"
 "MAM00096c"	"MAM00096"	""	"C02100"	""	""	""	""	""	""	"M00096"	"MNXM92760"	"m00096c"	"m00096c"
-"MAM00097s"	"MAM00097"	"gluala"	"C03740"	"HMDB06248"	"CHEBI:50619"	"440103"	""	""	""	"gluala"	"MNXM59390"	"m00097s"	"m00097s"
+"MAM00097s"	"MAM00097"	"gluala"	""	"HMDB06248"	"CHEBI:50619"	"440103"	""	""	""	"gluala"	"MNXM59390"	"m00097s"	"m00097s"
```
Comment (JonathanRob) [inline model/metabolites.tsv:186]: To address the current conflict with `develop`, could you please change `MAM00097s` (in the first column) to `MAM00097e`

File: model/metabolites.tsv [hunk 2]
```diff
@@ -1312,7 +1312,7 @@ mets	metsNoComp	metBiGGID	metKEGGID	metHMDBID	metChEBIID	metPubChemID	metLipidMa
 "MAM00810s"	"MAM00810"	""	"C05394"	""	"CHEBI:27453"	"440653"	""	""	"HC01440"	"HC01440"	"MNXM4339"	"m00810s"	"m00810s"
 "MAM00811c"	"MAM00811"	""	""	""	""	""	""	"CE4811"	""	"CE4811"	"MNXM164249"	"m00811c"	"m00811c"
 "MAM00812s"	"MAM00812"	""	"C05403"	""	""	""	""	""	"HC01446"	"HC01446"	"MNXM36578"	"m00812s"	"m00812s"
-"MAM00813p"	"MAM00813"	""	"C07297"	"HMDB02057"	"CHEBI:28542"	"441253"	""	""	""	"C07297"	"MNXM4346"	"m00813p"	"m00813p"
+"MAM00813p"	"MAM00813"	""	"C07297"	"HMDB02057"	"CHEBI:57291"	"441253"	""	""	""	"C07297"	"MNXM4346"	"m00813p"	"m00813p"
```
Comment (JonathanRob) [inline model/metabolites.tsv:1315]: To address the current conflict with `develop`, could you please change `MAM00813p` (in the first column) to `MAM00813x`

File: model/metabolites.tsv [hunk 3]
```diff
@@ -2674,10 +2674,10 @@ mets	metsNoComp	metBiGGID	metKEGGID	metHMDBID	metChEBIID	metPubChemID	metLipidMa
...
 "MAM01613s"	"MAM01613"	"core8"	""	""	""	""	""	""	""	"core8"	"MNXM16842"	"m01613s"	"m01613s"
-"MAM01614c"	"MAM01614"	"crtstrn"	""	""	"CHEBI:101161"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614c"	"m01614c"
-"MAM01614m"	"MAM01614"	"crtstrn"	""	""	"CHEBI:101161"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614m"	"m01614m"
-"MAM01614r"	"MAM01614"	"crtstrn"	""	""	"CHEBI:101161"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614r"	"m01614r"
-"MAM01614s"	"MAM01614"	"crtstrn"	""	""	"CHEBI:101161"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614s"	"m01614s"
+"MAM01614c"	"MAM01614"	"crtstrn"	""	""	"CHEBI:16827"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614c"	"m01614c"
+"MAM01614m"	"MAM01614"	"crtstrn"	""	""	"CHEBI:16827"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614m"	"m01614m"
+"MAM01614r"	"MAM01614"	"crtstrn"	""	""	"CHEBI:16827"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614r"	"m01614r"
+"MAM01614s"	"MAM01614"	"crtstrn"	""	""	"CHEBI:16827"	""	"LMST02030092"	""	""	"crtstrn"	"MNXM162953;MNXM859"	"m01614s"	"m01614s"
```
Comment (JonathanRob) [inline model/metabolites.tsv:2680]: To address the current conflict with `develop`, could you please change `MAM01614s` (in the first column) to `MAM01614e`

**Review by JonathanRob — CHANGES_REQUESTED** [review]
Looks good - thanks again for your help! I suggested a few changes you could make on your end to update the metabolite identifiers (which have changed on `develop` during your work). Once these changes are made, then it should help to solve the conflicts that currently exist between these branches.

File: model/metabolites.tsv (hunk 1 above)
Comment (ANiknejad) [reply model/metabolites.tsv]: done, please see https://github.com/SysBioChalmers/Human-GEM/pull/358/commits/041ad028d203e6015589d8e901f2bf0ea07bb0be

File: model/metabolites.tsv (hunk 2 above)
Comment (ANiknejad) [reply model/metabolites.tsv]: done, please see https://github.com/SysBioChalmers/Human-GEM/pull/358/commits/041ad028d203e6015589d8e901f2bf0ea07bb0be

File: model/metabolites.tsv (hunk 3 above)
Comment (ANiknejad) [reply model/metabolites.tsv]: done, please see https://github.com/SysBioChalmers/Human-GEM/pull/358/commits/041ad028d203e6015589d8e901f2bf0ea07bb0be

**Review by JonathanRob — APPROVED** [review]
Thanks, @ANiknejad. I _think_ it should now be OK to merge these into `develop` (using squash & merge, and --ours?) but I will ask for @mihai-sysbio's help here, since he is more experienced with git.

**Comment by JonathanRob** [comment]
@mihai-sysbio Ok, thanks. Yes, the squash portion makes sense, it was just the "favoring changes in this branch over those in `develop`" that I am hesitant about.

The lines/rows in `metabolites.tsv` in this branch that contain changes are correct, and should be favored. However, when I view the conflicts here in GitHub (using e.g. the `Resolve conflicts` button), it seems that in some cases it is including lines above/below those with changes (some of which are outdated and should not be favored). So I am uncertain if merging will also bring these "surrounding" lines along.

**Comment by mihai-sysbio** [comment]
It was unexpectedly tricky to resolve this - thanks for the heads up @JonathanRob. I ended up using `Resolve conflicts` and checking manually line-by-line which change should be taken in. The new changes in this PR were good to go. Some of the surrounding lines were caused by old compartment identifiers, which I cross-checked against the PR where these letters were last updated https://github.com/SysBioChalmers/Human-GEM/pull/341#issue-1084160677. 
The merge appears ready now, and as far as I can see all @ANiknejad's changes are still there. I'm leaving this to @JonathanRob, in case you want to have one last look.

**Comment by JonathanRob** [comment]
Great, thanks @mihai-sysbio for your help on this.

**Comment by haowang-bioinfo** [comment]
@ANiknejad sorry for the late response. Everything looks great with the excellent work of @mihai-sysbio and @JonathanRob

**Pull request merged**